src/Widgets/calendar/GridDay.vala

```diff
@@ -82,6 +89,38 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
     public void update_date (GLib.DateTime date) {
         this.date = date;
         label.label = date.get_day_of_month ().to_string ();
+
+        Widgets.CalendarModel.get_default ().events_added.connect (update_event_days);
```

Comment:
Is there a reason you're adding these here instead of in the construct block? Since this is a public method, this means these signals will possibly be reconnected over and over again

Reply:
This method is used in `Grid.vala` for the construction of each day. The idea here was, to connect to the signals for each day, and see if they have events or not and display the dots if they have events.

Reply:
Nevermind the above, I am dumb.
It's now moved to the construction block and works. Sorry.